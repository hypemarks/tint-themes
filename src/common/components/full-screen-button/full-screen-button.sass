@import '../../../styles/core/core'

.fullscreen
  position: fixed
  width: 100%
  z-index: 12
  top: 65px
  text-align: center

  &__button
    border-radius: 50px
    padding: 12px 30px
    background: $c-turquoise
    border-color: transparent
    color: $c-white
    box-shadow: 0 1px 1px 0 rgba(40,40,40,.3)
