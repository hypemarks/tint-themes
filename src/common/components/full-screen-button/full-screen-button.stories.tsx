import React from 'react';
import { Story, Meta } from '@storybook/react/types-6-0';
import { FullScreenButton } from './full-screen-button';

export default {
  title: 'Common/FullScreenButton',
  name: 'Full Screen Button',
  component: FullScreenButton,
} as Meta;

const Template: Story = args => <FullScreenButton {...args} />;

export const Basic = Template.bind({});
Basic.args = {};
