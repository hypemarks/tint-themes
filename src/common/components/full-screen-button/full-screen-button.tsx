import React, { FC, useEffect, useState } from 'react';
import './full-screen-button.sass';
import Button from '../button/button';

declare global {
  interface HTMLElement {
    msRequestFullscreen?: () => Promise<void>;
    mozRequestFullScreen?: () => Promise<void>;
    webkitRequestFullscreen?: () => Promise<void>;
  }
}

export const FullScreenButton: FC = () => {
  const [isFullScreen, setIsFullScreen] = useState(false);

  // switch instead of ifology?
  const toggleFullScreen = () => {
    const elem = document.documentElement;
    if (elem.requestFullscreen) {
      elem.requestFullscreen();
    } else if (elem.msRequestFullscreen) {
      elem.msRequestFullscreen();
    } else if (elem.mozRequestFullScreen) {
      elem.mozRequestFullScreen();
    } else if (elem.webkitRequestFullscreen) {
      elem.webkitRequestFullscreen();
    }
  };

  const fullScreenChangeHandler = () => {
    if (document.fullscreenElement) {
      setIsFullScreen(true);
    } else {
      setIsFullScreen(false);
    }
  };

  useEffect(() => {
    document.addEventListener('fullscreenchange', fullScreenChangeHandler);
    return () => {
      document.removeEventListener('fullscreenchange', fullScreenChangeHandler);
    };
  }, []);

  return !isFullScreen ? (
    <div className='fullscreen'>
      <Button className='fullscreen__button' onClick={toggleFullScreen}>
        Click to enable full screen
      </Button>
    </div>
  ) : null;
};
