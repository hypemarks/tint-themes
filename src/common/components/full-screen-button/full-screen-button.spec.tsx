import { render } from '@testing-library/react';
import React from 'react';
import { FullScreenButton } from './full-screen-button';

describe('Test <FullScreenButton /> Component', () => {
  it('should render snapshot', () => {
    const wrapper = render(<FullScreenButton />);
    expect(wrapper.asFragment()).toMatchSnapshot();
  });
});
