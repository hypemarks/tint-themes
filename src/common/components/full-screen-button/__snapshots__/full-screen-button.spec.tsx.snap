// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <FullScreenButton /> Component should render snapshot 1`] = `
<DocumentFragment>
  <div
    class="fullscreen"
  >
    <button
      class="fullscreen__button button"
      data-testid="button"
      tabindex="0"
      type="button"
    >
      Click to enable full screen
    </button>
  </div>
</DocumentFragment>
`;
